internal/data/resources/resourcerepository.go

```diff
@@ -47,46 +47,48 @@ func (r *Repo) Create(ctx context.Context, m *model.Resource, namespace string,
 	db := r.DB.Session(&gorm.Session{})
 	var result *model.Resource
 	err := r.handleSerializableTransaction(db, func(tx *gorm.DB) error {
+		// Copy the resource to avoid modifying the original, necessary for serialized transaction retries
+		resource := *m
```

Comment:
can `*m` can be a nil?

Reply:
Practically it will never be nil given the nature of what we need to accomplish with the create function - creating a resource from that model. Hypothetically, it could but it would be a bug. I'll add some nil checks just for completeness.